stdlib/fileutils/0/fileutils.rbs

```diff
@@ -0,0 +1,569 @@
+#     FileUtils.ln(targets, dir, **options)
+#     FileUtils.ln_s(target, link, **options)
+#     FileUtils.ln_s(targets, dir, **options)
+#     FileUtils.ln_sf(target, link, **options)
+#     FileUtils.cp(src, dest, **options)
+#     FileUtils.cp(list, dir, **options)
+#     FileUtils.cp_r(src, dest, **options)
+#     FileUtils.cp_r(list, dir, **options)
+#     FileUtils.mv(src, dest, **options)
+#     FileUtils.mv(list, dir, **options)
+#     FileUtils.rm(list, **options)
+#     FileUtils.rm_r(list, **options)
+#     FileUtils.rm_rf(list, **options)
+#     FileUtils.install(src, dest, **options)
+#     FileUtils.chmod(mode, list, **options)
+#     FileUtils.chmod_R(mode, list, **options)
+#     FileUtils.chown(user, group, list, **options)
+#     FileUtils.chown_R(user, group, list, **options)
+#     FileUtils.touch(list, **options)
+#
+# Possible `options` are:
+#
+# `:force`
+# :   forced operation (rewrite files if exist, remove directories if not empty,
+#     etc.);
+# `:verbose`
+# :   print command to be run, in bash syntax, before performing it;
+# `:preserve`
+# :   preserve object's group, user and modification time on copying;
+# `:noop`
+# :   no changes are made (usable in combination with `:verbose` which will
+#     print the command to run)
+#
+#
+# Each method documents the options that it honours. See also ::commands,
+# ::options and ::options_of methods to introspect which command have which
+# options.
+#
+# All methods that have the concept of a "source" file or directory can take
+# either one file or a list of files in that argument.  See the method
+# documentation for examples.
+#
+# There are some `low level' methods, which do not accept keyword arguments:
+#
+#     FileUtils.copy_entry(src, dest, preserve = false, dereference_root = false, remove_destination = false)
+#     FileUtils.copy_file(src, dest, preserve = false, dereference = true)
+#     FileUtils.copy_stream(srcstream, deststream)
+#     FileUtils.remove_entry(path, force = false)
+#     FileUtils.remove_entry_secure(path, force = false)
+#     FileUtils.remove_file(path, force = false)
+#     FileUtils.compare_file(path_a, path_b)
+#     FileUtils.compare_stream(stream_a, stream_b)
+#     FileUtils.uptodate?(file, cmp_list)
+#
+# ## module FileUtils::Verbose
+#
+# This module has all methods of FileUtils module, but it outputs messages
+# before acting.  This equates to passing the `:verbose` flag to methods in
+# FileUtils.
+#
+# ## module FileUtils::NoWrite
+#
+# This module has all methods of FileUtils module, but never changes
+# files/directories.  This equates to passing the `:noop` flag to methods in
+# FileUtils.
+#
+# ## module FileUtils::DryRun
+#
+# This module has all methods of FileUtils module, but never changes
+# files/directories.  This equates to passing the `:noop` and `:verbose` flags
+# to methods in FileUtils.
+module FileUtils
+  VERSION: String
+
+  type mode = Integer | String
+
+  type pathlist = String | Array[String]
```

Comment:
I guess it also accepts `_ToPath` and `string`.
https://github.com/ruby/rbs/blob/e7504e60517f3a4484794e4b0fa3d4b0b0641ae5/core/builtin.rbs#L25-L27

https://github.com/ruby/rbs/blob/e7504e60517f3a4484794e4b0fa3d4b0b0641ae5/core/builtin.rbs#L64


Because most `File` methods accept them.

Reply:
Thanks, it is a nice idea!

Also, it seems good to add the following type to `core/builtin.rbs`, what do you think?

```ruby
type path = string | _ToPath
```

Anyway, I will improve the signatures via new following types: 👍 

```
module FileUtils
  type path = string | _ToPath
  type pathlist = path | Array[path]
end
```

Reply:
Fixed via 929ecc8

Reply:
> Also, it seems good to add the following type to core/builtin.rbs, what do you think?



Personally I think the `path` type alias should not be defined to the top level. `path` alias is a domain-specific type, unlike `string` and so on. So I think it should be defined under a namespace, and I think `File::path` is appropriate.


And we need to reconsider the other type aliases and interfaces, such as `_ToPath`, `_Reader` and so on.

So I'd like to keep them in this pull request, and move them under a namespace by another pull request.

Reply:
That makes sense. 👍